src/main/io/vtx.c

```diff
@@ -80,30 +82,40 @@ void vtxProcess(timeUs_t currentTimeUs)
         if (currentTimeUs > lastCycleTimeUs + VTX_PARAM_CYCLE_TIME_US) {
             switch (currentSchedule) {
             case VTX_PARAM_BANDCHAN:
-                if (vtxSettingsConfig()->band) {
-                    uint8_t vtxBand;
-                    uint8_t vtxChan;
-                    if (vtxCommonGetBandAndChannel(&vtxBand, &vtxChan)) {
-                        if (vtxSettingsConfig()->band != vtxBand || vtxSettingsConfig()->channel != vtxChan) {
-                            vtxCommonSetBandAndChannel(vtxSettingsConfig()->band, vtxSettingsConfig()->channel);
+                if(!ARMING_FLAG(ARMED)) {
+                    if (vtxSettingsConfig()->band) {
+                        uint8_t vtxBand;
+                        uint8_t vtxChan;
+                        if (vtxCommonGetBandAndChannel(&vtxBand, &vtxChan)) {
+                            if (vtxSettingsConfig()->band != vtxBand || vtxSettingsConfig()->channel != vtxChan) {
+                                vtxCommonSetBandAndChannel(vtxSettingsConfig()->band, vtxSettingsConfig()->channel);
+                            }
                         }
-                    }
 #if defined(VTX_SETTINGS_FREQCMD)
-                } else {
-                    uint16_t vtxFreq;
-                    if (vtxCommonGetFrequency(&vtxFreq)) {
-                        if (vtxSettingsConfig()->freq != vtxFreq) {
-                            vtxCommonSetFrequency(vtxSettingsConfig()->freq);
+                    } else {
+                        uint16_t vtxFreq;
+                        if (vtxCommonGetFrequency(&vtxFreq)) {
+                            if (vtxSettingsConfig()->freq != vtxFreq) {
+                                vtxCommonSetFrequency(vtxSettingsConfig()->freq);
+                            }
                         }
-                    }
 #endif
+                    }
                 }
                 break;
             case VTX_PARAM_POWER: ;
                 uint8_t vtxPower;
+                uint8_t newPower;
                 if (vtxCommonGetPowerIndex(&vtxPower)) {
-                    if (vtxSettingsConfig()->power != vtxPower) {
-                        vtxCommonSetPowerByIndex(vtxSettingsConfig()->power);
+                    if (!ARMING_FLAG(ARMED)) {
```

Comment:
As discussed in #4334, I think a timeout would be good because you probably don't want to switch power if you just arm/disarm quickly to check that props are spinning. I do that all the time and I guess others too (see Tom Smith videos, he does it every time).